iceoryx2/src/service/static_config/mod.rs

```diff
@@ -24,13 +24,16 @@ pub mod publish_subscribe;
 /// and the type variant
 pub mod message_type_details;
 
-use crate::service::messaging_pattern::MessagingPattern;
+pub mod messaging_pattern;
+
 use iceoryx2_bb_log::fatal_panic;
 use iceoryx2_cal::hash::Hash;
 use serde::{Deserialize, Serialize};
 
 use crate::config;
 
+use self::messaging_pattern::MessagingPattern;
```

Comment:
Is this different from `crate::service::messaging_pattern::MessagingPattern`?

Reply:
Yes, this is the generic plain MessagingPattern enum. The type `MessagingPattern` is again available for `static_config` and `dynamic_config` which then contains the messaging pattern specific dynamic or static configuration.